tools/PsesPsClient/PsesPsClient.psm1

```diff
@@ -0,0 +1,314 @@
+    param(
+        [Parameter(Mandatory)]
+        [string]
+        $PipeName
+    )
+
+    $psesIdx = $PipeName.IndexOf('PSES')
+    if ($psesIdx -gt 0)
+    {
+        $PipeName = $PipeName.Substring($psesIdx)
+    }
+
+    $pipe = [PsesPsClient.LspPipe]::Create($PipeName)
+    $pipe.Connect()
+    return $pipe
+}
+
+function Send-LspInitializeRequest
+{
+    [OutputType([PsesPsClient.LspRequest])]
+    param(
+        [Parameter(Position = 0, Mandatory)]
+        [PsesPsClient.LspPipe]
+        $Pipe,
+
+        [Parameter()]
+        [int]
+        $ProcessId = $PID,
+
+        [Parameter()]
+        [string]
+        $RootPath = (Get-Location),
+
+        [Parameter()]
+        [string]
+        $RootUri,
+
+        [Parameter()]
+        [Microsoft.PowerShell.EditorServices.Protocol.LanguageServer.ClientCapabilities]
+        $ClientCapabilities = ([Microsoft.PowerShell.EditorServices.Protocol.LanguageServer.ClientCapabilities]::new()),
+
+        [Parameter()]
+        [hashtable]
+        $InitializeOptions = $null
+    )
+
+    $parameters = [Microsoft.PowerShell.EditorServices.Protocol.LanguageServer.InitializeParams]@{
+        ProcessId = $ProcessId
+        Capabilities = $ClientCapabilities
+        InitializeOptions = $InitializeOptions
+    }
+
+    if ($RootUri)
+    {
+        $parameters.RootUri = $RootUri
+    }
+    else
+    {
+        $parameters.RootPath = $RootPath
+    }
+
+    return $Pipe.WriteRequest('initialize', $parameters)
+}
+
+function Send-LspShutdownRequest
+{
+    [OutputType([PsesPsClient.LspRequest])]
+    param(
+        [Parameter(Position = 0, Mandatory)]
+        [PsesPsClient.LspPipe]
+        $Pipe
+    )
+
+    $Pipe.WriteRequest('shutdown', $null)
+}
+
+function Unsplat
+{
+    param(
+        [string]$Prefix,
+        [hashtable]$SplatParams)
+
+    $sb = New-Object 'System.Text.StringBuilder' ($Prefix)
+
+    foreach ($key in $SplatParams.get_Keys())
+    {
+        $val = $SplatParams[$key]
+
+        if (-not $val)
+        {
+            continue
+        }
+
+        $null = $sb.Append(" -$key")
+
+        if ($val -is [switch])
+        {
+            continue
+        }
+
+        if ($val -is [array])
+        {
+            $null = $sb.Append(' @(')
+            for ($i = 0; $i -lt $val.Count; $i++)
+            {
+                $null = $sb.Append("'").Append($val[$i]).Append("'")
+                if ($i -lt $val.Count - 1)
+                {
+                    $null = $sb.Append(',')
+                }
+            }
+            $null = $sb.Append(')')
+            continue
+        }
+
+        if ($val -is [version])
+        {
+            $val = [string]$val
+        }
+
+        if ($val -is [string])
+        {
+            $null = $sb.Append(" '$val'")
+            continue
+        }
+
+        throw "Bad value '$val' of type $($val.GetType())"
+    }
+
+    return $sb.ToString()
+}
+
+$script:Random = [System.Random]::new()
+function Get-RandomHexString
```

Comment:
Get-Random doesn't do anything here, huh?

Reply:
Nah, it only gives back an `int`. That's probably practically enough, but I've already done this (I've written this function a few times, for tests in PowerShell for example).

Really we should see if `Get-Random` is worth adding a new parameter set to